statsmodels/tsa/_stl.pyx

```diff
@@ -0,0 +1,680 @@
+        provided, uses the smallest odd integer > period.
+    seasonal_deg: int, optional
+        Degree of seasonal LOESS. 0 (constant) or 1 (constant and trend)
+    trend_deg: int, optional
+        Degree of trend LOESS. 0 (constant) or 1 (constant and trend)
+    low_pass_deg: int, optional
+        Degree of low pass LOESS. 0 (constant) or 1 (constant and trend)
+    robust: bool, optional
+        Flag indicating whether to use a weighted version that is robust to
+        some forms of outliers
+    seasonal_jump: int, optional
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every seasonal_jump points and values between
+        the two are linearly interpolated. Higher values reduce estimation time
+    trend_jump: int, optional
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every trend_jump points and values between
+        the two are linearly interpolated. Higher values reduce estimation time
+    low_pass_jump: int, optional
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every low_pass_jump points and values between
+        the two are linearly interpolated. Higher values reduce estimation time
+
+    Notes
+    -----
+    Derived from the NETLIB fortran written by [1]_.  The original code
+    contains a bug that appears in the determination of the median that is
+    used in the robust weighting. This version matches the fixed version that
+    uses a correct partitioned sort to determine the median.
+
+    Examples
+    --------
+    The original example uses STL to decompos CO2 data into level, season and a
+    residual.
+
+    Start by aggregating to monthly, and filling any missing values
+
+    >>> from statsmodels.datasets import co2
+    >>> import matplotlib.pyplot as plt
+    >>> from pandas.plotting import register_matplotlib_converters
+    >>> register_matplotlib_converters()
+    >>> data = co2.load(True).data
+    >>> data = data.resample('M').mean().ffill()
+
+    The period (12) is automatically detected from the data's frequency ('M').
+
+    >>> from statsmodels.tsa.seasonal import STL
+    >>> res = STL(data).fit()
+    >>> res.plot()
+
+    See Also
+    --------
+    statsmodels.tsa.seasonal.STLResults
+    statsmodels.tsa.seasonal.seasonal_decompose
+
+    References
+    ----------
+    .. [1] R. B. Cleveland, W. S. Cleveland, J.E. McRae, and I. Terpenning
+        (1990) STL: A Seasonal-Trend Decomposition Procedure Based on LOESS.
+        Journal of Official Statistics, 6, 3-73.
+    """
+    cdef object endog
+    cdef Py_ssize_t nobs
+    cdef int period, seasonal, trend, low_pass, seasonal_deg, trend_deg
+    cdef int low_pass_deg, low_pass_jump, trend_jump, seasonal_jump
+    cdef bint robust, _use_rw
+    cdef double[::1] _ya, _trend, _season, _rw
+    cdef double[:, ::1] _work
```

Comment:
unless there is a compelling reason to use `double`, `float64_t` avoids some ambiguity

Reply:
Probably better to allow platforms to define double the want they prefer.